vllm/v1/core/sched/scheduler.py

```diff
@@ -751,17 +835,33 @@ def update_from_output(
                         new_logprobs=new_logprobs,
                         new_prompt_logprobs_tensors=prompt_logprobs_tensors,
                         stop_reason=request.stop_reason,
-                        events=request.take_events()))
+                        events=request.take_events(),
+                        kv_transfer_params=kv_transfer_params,
+                    ))
+
             else:
                 # Invariant: EngineCore returns no partial prefill outputs.
                 assert not prompt_logprobs_tensors
 
             if not stopped:
                 new_running.append(request)
-
-        # Return the cached request data to the queue so they can be reused.
+            else:
+                stopped_set.add(request.request_id)
+
+        # P/D: update recv and send status from last step.
+        for req_id in (model_runner_output.finished_recving or []):
+            logger.debug("Finished recving KV transfer for request %s", req_id)
+            self.finished_recving_KV_req_ids.add(req_id)
```

Comment:
probably more efficient to do
```suggestion
        self.finished_recving_KV_req_ids.update(model_runner_output.finished_recving or ())
```